server/src/main/java/keywhiz/service/daos/SecretSeriesMapper.java

```diff
@@ -50,6 +60,25 @@ public SecretSeries map(SecretsRecord r) {
         r.getCurrent());
   }
 
+  private String getOwner(SecretsRecord r) {
+    Long ownerId = r.getOwner();
+    if (ownerId == null) {
+      return null;
+    }
+
+    Optional<Group> maybeGroup = groupDAO.getGroupById(ownerId);
+    if (maybeGroup.isEmpty()) {
+      throw new IllegalStateException(
+          String.format(
+              "Unable to find owner for secret [%s] (ID %s): group ID %s not found",
+              r.getName(),
+              r.getId(),
+              ownerId));
+    }
+
+    return maybeGroup.get().getName();
+  }
```

Comment:
We support missing (null) owners, but we don't support references to owners which don't exist. Normally we would enforce this at the database level via a foreign key constraint but our testing setup's use of truncate precludes this (and I haven't had time to dig in to find a better solution).

Reply:
Right, for backwards compatibility we allow owners to be null since we can't atomically update Keywhiz, the Keywhiz database, and Secrets Authority together. However, going forward it may make sense to lock that down since it won't make sense to create secrets without owners (I think), at least from the Console/Secrets Authority path (note to self: I should figure out if there are other secret creation paths which we'll need to support and what changes may be necessary there [maybe add a counter every time a secret is created without an owner?]). But, we also *do not* allow you to specify an owner which doesn't exist. I *think* it makes sense for the caller to ensure that the owner exists beforehand by making the appropriate group fetch/create calls rather than have the secret creation API silently create a group if none exists (also the problem with that option is that we won't have all the fields necessary to properly create the group if we're limited to whatever the secret creation API accepts as parameters).